common/helpers/field/is-allowed-dependent.js

```diff
@@ -0,0 +1,40 @@
+const { intersection } = require('lodash')
+
+function isAllowedDependent(fields, key, values) {
+  const field = fields[key]
+
+  if (!field) {
+    return false
+  }
+
+  // set up dependent to an object of field and value
+  if (typeof field.dependent === 'string') {
+    field.dependent = {
+      field: field.dependent,
+      value: true,
+    }
+  }
+
+  if (!field.dependent) {
```

Comment:
Couldn't this check be done before the string to object coercion

Reply:
Yep, I'll move it and make sure the tests still pass.